File: reactivando-backend/src/main/java/com/reactivando/futuro/service/VacanteService.java
package com.reactivando.futuro.service;

import com.reactivando.futuro.dto.vacante.VacanteRequestDTO;
import com.reactivando.futuro.dto.vacante.VacanteResponseDTO;
import com.reactivando.futuro.entity.EstadoVacante;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;
import java.util.List;

public interface VacanteService {
    VacanteResponseDTO crearVacante(String correoUsuario, VacanteRequestDTO requestDTO);
    VacanteResponseDTO obtenerPorId(Long id);
    Page<VacanteResponseDTO> listarVacantes(Pageable pageable);
    Page<VacanteResponseDTO> buscarVacantes(
            String titulo,
            String ciudad,
            String tipoContrato,
            String nivelEstudio,
            BigDecimal salarioMin,
            BigDecimal salarioMax,
            EstadoVacante estado,
            Pageable pageable
    );
    Page<VacanteResponseDTO> listarPorEmpresa(Long empresaId, Pageable pageable);
    List<VacanteResponseDTO> listarTodasPorEmpresa(Long empresaId);
    VacanteResponseDTO actualizarVacante(Long id, String correoUsuario, VacanteRequestDTO requestDTO);
    VacanteResponseDTO cambiarEstado(Long id, String correoUsuario, EstadoVacante nuevoEstado);
    void eliminarVacante(Long id, String correoUsuario);
}

File: reactivando-frontend/src/services/vacanteService.js
import api from './api';

export const vacanteService = {
  listarVacantes: async (page = 0, size = 10) => {
    const response = await api.get(`/vacantes?page=${page}&size=${size}`);
    return response.data;
  },

  buscarVacantes: async (params = {}, page = 0, size = 10) => {
    const queryParams = new URLSearchParams({ page, size, ...params }).toString();
    const response = await api.get(`/vacantes/buscar?${queryParams}`);
    return response.data;
  },

  obtenerPorId: async (id) => {
    const response = await api.get(`/vacantes/${id}`);
    return response.data;
  },

  listarPorEmpresa: async (empresaId, page = 0, size = 10) => {
    const response = await api.get(`/vacantes/empresa/${empresaId}?page=${page}&size=${size}`);
    return response.data;
  },

  crearVacante: async (vacanteData) => {
    const response = await api.post('/vacantes', vacanteData);
    return response.data;
  },

  actualizarVacante: async (id, vacanteData) => {
    const response = await api.put(`/vacantes/${id}`, vacanteData);
    return response.data;
  },

  cambiarEstado: async (id, estado) => {
    const response = await api.patch(`/vacantes/${id}/estado?estado=${estado}`);
    return response.data;
  },

  eliminarVacante: async (id) => {
    await api.delete(`/vacantes/${id}`);
  },
};
